rest/tests/test_rest_authorization.py

```diff
@@ -0,0 +1,476 @@
+                               expected_status=HTTP_403_FORBIDDEN)
+
+        self.logout()
+        self.login(self.curator_user.username)
+
+        # curator does have view_repo but nobody is allowed
+        # to delete repositories
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_405_METHOD_NOT_ALLOWED)
+
+        # switch to user with no permissions for repo
+        self.logout()
+        self.login(self.user_norepo.username)
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_403_FORBIDDEN)
+
+        # as anonymous
+        self.logout()
+        self.delete_repository(self.repo.slug,
+                               expected_status=HTTP_403_FORBIDDEN)
+
+    def test_repository_get(self):
+        """Test list and view repository"""
+
+        # switch to add_repo_user
+        self.logout()
+        self.login(self.add_repo_user.username)
+
+        # add_repo_user shouldn't be able to see any repositories yet
+        self.get_repository(self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        repos = self.get_repositories()
+        self.assertEqual(0, repos['count'])
+
+        # add_repo_user creates a repository
+        new_repo_slug = self.create_repository()['slug']
+
+        self.get_repository(new_repo_slug)
+        self.get_repository(self.repo.slug, expected_status=HTTP_403_FORBIDDEN)
+        repos = self.get_repositories()
+        self.assertEqual(1, repos['count'])
+
+        # switch to curator_user
+        self.logout()
+        self.login(self.curator_user.username)
+
+        self.assertEqual(1, self.get_repositories()['count'])
+        self.get_repository(new_repo_slug,
+                            expected_status=HTTP_403_FORBIDDEN)
+        self.get_repository(self.repo.slug)
+
+        # switch to user_norepo
+        self.logout()
+        self.login(self.user_norepo.username)
+
+        self.assertEqual(0, self.get_repositories()['count'])
+        self.get_repository(self.repo.slug,
+                            expected_status=HTTP_403_FORBIDDEN)
+
+        # as anonymous
+        self.logout()
+        self.get_repositories(expected_status=HTTP_403_FORBIDDEN)
+        self.get_repository(self.repo.slug,
+                            expected_status=HTTP_403_FORBIDDEN)
+
+    def test_vocabulary_create(self):
+        """Test create vocabulary"""
+        self.logout()
+        self.login(self.curator_user.username)
+
+        self.create_vocabulary(self.repo.slug)
+        self.assertEqual(1, self.get_vocabularies(self.repo.slug)['count'])
+
+        # login as author which doesn't have manage_taxonomy permissions
+        self.logout()
+        self.login(self.author_user.username)
+
+        vocab_dict = dict(self.DEFAULT_VOCAB_DICT)
+        vocab_dict['name'] = 'other_name'  # prevent name collision
+
+        # author does not have create access, but we haven't disabled this yet
```

Comment:
These tests were copied from my taxonomy authorizations branch so I could get a better picture what the state of taxonomy authorization is in this PR. If these tests break it signals a change of behavior which may be important to us even if it's toward a more correct behavior. In any case they will change assertions in the next PR toward correct behavior.